### PR TITLE
fix: disambiguate generated integration docs

### PR DESCRIPTION
**Pull request by notgitika**
## Summary
- label Strands and Vercel AI integration entries separately
- emit stable explicit anchors for integration entries
- normalize generated AsciiDoc for AWS style and China rebranding checks
- add a regression test for duplicate integration names

Addresses generated documentation feedback from Amazon CR-290947859.

## Testing
- `node --test scripts/extract-api-model.test.mjs`
- repository pre-commit checks: 23 test files / 617 tests, lint, format, and type-check
- regenerated the v0.4.1 TypeScript reference

### CODE REVIEW

**Comment by notgitika** [comment]
@strandly-the-agent review

File: scripts/render_adoc.py [hunk 1]
```diff
@@ -55,15 +55,44 @@
...
+    )
+    return re.sub(r"\bAWS\b", "{aws}", text)
+
+
+def normalize_param_description(text):
+    """Normalize recurring parameter-description style issues."""
+    text = normalize_style(text).strip()
+    substitutions = (
+        (r"^Optional\b", "An optional"),
+        (r"^(?:\{aws\}|AWS)\s+region\b", "The {aws} Region"),
```
Comment (strandly-the-agent) [inline scripts/render_adoc.py:108]: 🟡 This unconditional article makes plural and mass-noun descriptions ungrammatical. I ran the full generator and found 10 current examples, including `An optional parameters specifying which session to query`, `An optional filtering and pagination parameters`, `An optional task metadata`, and `An optional server options.` A wording that works for singular and plural inputs avoids generating broken public prose:

```suggestion
        (r"^Optional\b", "The optional"),
```

File: scripts/extract-api-model.test.mjs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    kind,
+    name,
+    sources: [{ fileName }],
+    signatures: kind === 64 ? [{ parameters: [], type: { name: 'void' } }] : undefined,
+    children: [],
+  }
+}
+
+test('integration entries have distinct labels and anchors', () => {
+  const doc = {
```
Comment (strandly-the-agent) [inline scripts/extract-api-model.test.mjs:18]: 🟡 This regression test is never exercised by the repository's standard gates. `npm test` only includes `src/**/__tests__/**/*.test.{ts,tsx}` and passed 617 tests without this file; `ci`, coverage, and the pre-commit hook all flow through the same package scripts. Please wire `node --test scripts/*.test.mjs` into a standard package/CI command so a regression actually fails validation.

**Review by strandly-the-agent — CHANGES_REQUESTED** [review]
🔴 **Changes requested:** the generator currently emits malformed prose, and the new regression test is not part of any standard test/CI command.

✅ Reviewed `b13dfc4ec4bdf8efc6b96ff4ab004b60306f9896` · full TypeDoc → model → AsciiDoc pipeline generated 5 files · standalone test 1/1 · unit 23 files/617 tests · lint/format/types/build passed.

⚠️ The PR rollup is red on `npm Audit`; this change adds no dependencies, so I did not attribute that failure to the diff.

<details><summary>Review breakdown</summary>

- Correctness/docs: reproduced 10 malformed parameter descriptions in current generated output.
- Test quality: confirmed `npm test` excludes `scripts/extract-api-model.test.mjs`; only its direct command runs it.
- Adversarial: anchor-collision and inline-code cases were synthetic with no current source impact, so I did not post them.
- API/LLM context: not applicable; this changes documentation tooling, not the public SDK API or model-facing context.
- No API-review label is required.

No additional questions.
</details>

File: scripts/render_adoc.py (hunk 1 above)
Comment (notgitika) [reply scripts/render_adoc.py:108]: Fixed in 788c651. The renderer now uses "The optional" for all Optional descriptions, and the regression test covers singular, plural, and mass-noun examples.

File: scripts/extract-api-model.test.mjs (hunk 2 above)
Comment (notgitika) [reply scripts/extract-api-model.test.mjs:18]: Fixed in 788c651. Added the test:docs package target and a PR Validation step that runs the extractor and renderer regression tests.

**Review by nborges-aws — APPROVED** [review]
